### PR TITLE
DBZ-3293 Add signal table automatically to include list

### PR DESCRIPTION
**Pull request by ggaborg**

### CODE REVIEW

**Comment by ggaborg** [comment]
MySQL and SQL Server tests are passing locally

**Comment by Naros** [comment]
Hi @ggaborg, do you intend to make the same test changes in the Vitess and Db2 connector repositories?

**Comment by ggaborg** [comment]
Yes, I was looking at it.

**Comment by Naros** [comment]
@ggaborg we don't need to wait for this PR, you can build the 2.0.0-SNAPSHOT locally based on your changes here and then make the necessary changes to Db2 and Vitess that pull in that snapshot version and test.  We'll merge all PRs at the same time so we guarantee that all repositories are aligned.

**Comment by ggaborg** [comment]
@Naros, correct, this is what I am doing. I just didn't know this procedure of multiple PR's

**Comment by ggaborg** [comment]
@Naros, PRs for DB2 and Vitess are ready.

**Review by Naros — APPROVED** [review]
Overall this LGTM @ggaborg, great work.  I've restarted the MySQL job to see if it passes.  The others are either unrelated or will be fixed when their corresponding sibling PRs are merged in their respective repositories. If MySQL passes I think we can merge this and the related PRs.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java [hunk 1]
```diff
@@ -57,7 +57,16 @@ else if (collectionExcludeList != null && !collectionExcludeList.trim().isEmpty(
             collectionFilter = (id) -> true;
         }
         Predicate<CollectionId> isNotBuiltIn = this::isNotBuiltIn;
-        this.collectionFilter = isNotBuiltIn.and(collectionFilter);
+        Predicate<CollectionId> finalCollectionFilter = isNotBuiltIn.and(collectionFilter);
+        String signalDataCollection = config.getString(MongoDbConnectorConfig.SIGNAL_DATA_COLLECTION);
+        if (signalDataCollection != null) {
+            CollectionId signalDataCollectionId = CollectionId.parse("UNUSED", signalDataCollection);
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java:63]: Could you please extend the `CollectionId` contract so no dummy value is needed to be passed?

**Review by jpechane — COMMENTED** [review]
@ggaborg Great work. I've left one comment related to MongoDB.
I understand that the feature is now more or less tested via integration tests. Yes I'd like to ask you to find unit tests for filters where exists and add cases for signalling collections/tables there too. After that we are good to merge this.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java (hunk 1 above)
Comment (ggaborg) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java:63]: What exactly are you thinking on? The CollectionId constructor expects three non null strings, provided that assertions are enabled.

I just took as example this line in the MongoDbIncrementalSnapshotChangeEventSource constructor:
        this.signallingCollectionId = connectorConfig.getSignalingDataCollectionId() == null ? null
                : CollectionId.parse("UNUSED", connectorConfig.getSignalingDataCollectionId());

Do you want to fix also this place?

Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java:63]: @ggaborg I see, then please create a follow-up Jira to get rid off both instances in a separate PR later.

Comment (ggaborg) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java:63]: @jpechane, created https://issues.redhat.com/browse/DBZ-5575 "Get rid of dummy value used to parse mongodb CollectionId"

**Comment by ggaborg** [comment]
@Naros, @jpechane can you squash on merge please? Because I have a bad commit message...

**Comment by Naros** [comment]
Hi @ggaborg we can do that or you can rebase the branch using `git rebase -i HEAD~1`, mark the latest commit to be `reworded`, modify the commit message, and the force push to your remote branch.  Either works.

**Comment by jpechane** [comment]
@ggaborg Thanks, I've squashed the commits. The PR will be merged after the test completion.

**Comment by jpechane** [comment]
@ggaborg Applied, thanks!

**Pull request merged**